src/plugins/data/server/search/collectors/usage.ts

```diff
@@ -38,44 +40,53 @@ export interface SearchUsage {
   trackSuccess(duration: number): Promise<void>;
 }
 
-export function usageProvider(core: CoreSetup): SearchUsage {
+export function usageProvider(
+  core: CoreSetup,
+  initializerContext: PluginInitializerContext<ConfigSchema>
+): SearchUsage {
   const getTracker = (eventType: keyof Usage) => {
     return async (duration?: number) => {
-      const repository = await core
-        .getStartServices()
-        .then(([coreStart]) => coreStart.savedObjects.createInternalRepository());
+      const config = await initializerContext.config
+        .create<ConfigSchema>()
+        .pipe(first())
+        .toPromise();
+      if (config.search.usageTelemetry.enabled) {
+        const repository = await core
+          .getStartServices()
+          .then(([coreStart]) => coreStart.savedObjects.createInternalRepository());
 
-      let attributes: Usage;
-      let doesSavedObjectExist: boolean = true;
+        let attributes: Usage;
```

Comment:
Neat. is it spacing / tab issue or my github diff checker is bad :|

Reply:
it is adjust indentation for 2 spaces. 

![image](https://user-images.githubusercontent.com/33105471/162114089-b611576e-b8fa-485b-b6c9-5610d4c457f4.png)